packages/polling-controller/src/PollingController.test.ts

```diff
@@ -0,0 +1,214 @@
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      expect(() => {
+        controller.stop('potato');
+      }).toThrow('pollingToken not found');
+      controller.stopAll();
+    });
+  });
+  describe('poll', () => {
+    it('should call executePoll if polling', async () => {
+      jest.useFakeTimers();
+
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      expect(executePollMock).toHaveBeenCalledTimes(2);
+    });
+    it('should continue calling executePoll when start is called again with the same networkClientId', async () => {
+      jest.useFakeTimers();
+
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      controller.start('mainnet');
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      expect(executePollMock).toHaveBeenCalledTimes(2);
+      controller.stopAll();
+    });
+    it('should publish polligComplete when stop is called', async () => {
+      jest.useFakeTimers();
+      const pollingComplete: any = jest.fn();
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const name = 'PollingController';
+
+      const mockMessenger = new ControllerMessenger<
+        any,
+        PollingCompleteType<typeof name>
+      >();
+
+      mockMessenger.subscribe(`${name}:pollingComplete`, pollingComplete);
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name,
+        state: { foo: 'bar' },
+      });
+      const pollingToken = controller.start('mainnet');
+      controller.stop(pollingToken);
+      expect(pollingComplete).toHaveBeenCalledTimes(1);
+    });
+  });
+  describe('multiple networkClientIds', () => {
+    it('should poll for each networkClientId', async () => {
```

Comment:
Nit: we could check the mock calls in this test to ensure the correct ids were passed in as well

Reply:
done here: https://github.com/MetaMask/core/pull/1703/commits/40ed0b5af488037745b6158a08cab3f0488145d8